app/src/main/java/com/asfoundation/wallet/iab/presentation/BonusScreen.kt

```diff
@@ -0,0 +1,169 @@
+    label = "ColorState"
+  )
+
+  val rotationState by animateFloatAsState(
+    targetValue = if (isExpanded) -180F else 0F,
+    label = "RotationState"
+  )
+
+  Column(
+    modifier = modifier,
+  ) {
+    Row(
+      verticalAlignment = Alignment.CenterVertically
+    ) {
+      if (bonusAvailable) {
+        Animation(
+          modifier = Modifier.size(20.dp),
+          animationRes = R.raw.iab_gift,
+          iterations = LottieConstants.IterateForever,
+        )
+      } else {
+        Image(
+          modifier = Modifier.size(20.dp),
+          imageVector = getIcGiftDisabled(),
+          contentDescription = null,
+        )
+      }
+      Text(
+        modifier = Modifier
+          .padding(start = 4.dp)
+          .fillMaxWidth()
+          .weight(1f),
+        text = bonusInfoData.bonusText,
+        style = IAPTheme.typography.bodyMedium,
+        color = if (bonusAvailable) IAPTheme.colors.onPrimary else IAPTheme.colors.disabledBColor
+      )
+      if (bonusAvailable) {
+        Image(
+          modifier = Modifier
+            .padding(start = 4.dp)
+            .size(12.dp)
+            .rotate(rotationState),
+          imageVector = getDownArrow(colorState),
+          contentDescription = null
+        )
+      }
+    }
+    AnimatedVisibility(isExpanded && bonusAvailable) {
+      Column(
+        modifier = Modifier.padding(top = 8.dp)
+      ) {
+        BonusDetails(
+          modifier = Modifier.padding(vertical = 8.dp),
+          name = "Promo code Perk",
+          value = bonusInfoData.promoCodePerkValue,
+        )
+        BonusDetails(
+          modifier = Modifier.padding(vertical = 8.dp),
+          name = "Gamification",
+          value = bonusInfoData.gamificationValue,
+        )
+        Text(
+          modifier = Modifier
+            .padding(top = 14.dp)
+            .clickable(onClick = onPromoCodeAvailableClick),
+          text = "Do you have a promo code?",
+          style = IAPTheme.typography.bodyMedium,
+          color = IAPTheme.colors.secondary,
+        ) //TODO hardcoded text
+      }
+    }
+  }
+}
+
+@Composable
+private fun BonusDetails(
+  modifier: Modifier = Modifier,
+  name: String,
+  value: String,
+) {
+  Row(modifier = modifier) {
+    Text(
+      modifier = Modifier
+        .fillMaxWidth()
+        .weight(1f),
+      text = name,
+      style = IAPTheme.typography.bodyMedium,
+      color = IAPTheme.colors.smallText,
+    )
+    Text(
+      text = value,
+      style = IAPTheme.typography.bodyMedium,
+      color = IAPTheme.colors.smallText,
+    )
+  }
+}
+
+@PreviewAll
+@Composable
+private fun PurchaseInfoPreview(
+  @PreviewParameter(IsBonusExpandedProvider::class) isExpanded: Boolean
+) {
+  IAPTheme {
+    BonusInfo(
+      isExpanded = isExpanded,
+      bonusInfoData = emptyBonusInfoData,
+      bonusAvailable = Random.nextBoolean(),
+      onPromoCodeAvailableClick = {},
```

Comment:
Don't forget to implement the test Tag IDs on everything that has a click, so we are ready for testing.